subtests/docker_cli/systemd/systemd.py

```diff
@@ -0,0 +1,87 @@
+"""
+Summary
+-------
+Testing basic container as systemd service
+
+Operational Summary
+---------------------
+This is to test container as systemd service using a predefined unit file
+(``p4321.service``) and a ``script p4321-server.py``. An image is built with
+this script added. Systemd starts a container using this image & the container
+writes current time to port ``4321``. From host, the socket is read and the
+value is checked to see if it is a digit. Finally, container is stopped
+through systemd and system is cleaned up.
+
+"""
+
+import shutil
+import time
+import socket
+from os.path import exists
+from os import unlink
+from autotest.client import utils
+from autotest.client.shared.utils import is_port_free
+from dockertest.subtest import Subtest
+from dockertest.dockercmd import DockerCmd
+from dockertest.images import DockerImages
+from dockertest.output import mustpass
+from dockertest.xceptions import DockerTestError
+
+
+class systemd(Subtest):
+
+    def initialize(self):
+        super(systemd, self).initialize()
+        self.stuff['systemd_touchstone'] = False
+        unit_file_srcpath = '{}{}'.format(self.bindir, '/p4321.service')
+        shutil.copyfile(unit_file_srcpath, self.config['sysd_unitf_dest'])
+        dkrcmd = DockerCmd(self, 'build', ['--force-rm -t',
+                                           self.config['name'], self.bindir])
+        mustpass(dkrcmd.execute())
+
+    def run_once(self):
+        host = 'localhost'
+        port = 4321
+        self.sysd_action('daemon-reload')
+        self.sysd_action('start', self.config['sysd_unit_file'])
+        while is_port_free(port, host):
+            time.sleep(0.1)
```

Comment:
Just realized, this could be an infinite loop e.g. something funky happens with docker/networking.  It will eventually exhaust the autotest-level timeout - a condition which means the test gets killed w/o running ``cleanup()`` (IIRC).  Maybe trying 1000 times (10-seconds), then barfing would be better.  @edsantiago  what do you think?